### PR TITLE
feat(angular): omit output related imports for components with no event

### LINKED ISSUE #331: feat(angular): omit output related imports for components with no event #330
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil-ds-output-targets/issues) that already report this problem, without success.

### Describe the Feature Request

This is valid only for the `angular-output-target` package.

Don't provide import entries that are not required for components to work.
Components that do not have any events does not require to import these 2 entries that currently are always delivered:
`import { EventEmitter } from '@angular/core';`
`import { proxyOutputs } from './angular-component-lib/utils';`

### Describe the Use Case

This feature will solve problem where TypeScript compiler complains about unused locals (`"noUnusedLocals": true`) and any other static code analyze tool that has ability to find unused imports (eg. eslint `unused-imports/no-unused-imports`).

### Describe Preferred Solution

_No response_

### Describe Alternatives

_No response_

### Related Code

https://github.com/ionic-team/stencil-ds-output-targets/pull/330

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by KariiO**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`npm run build`) was run locally for affected output targets
- [X] Tests (`npm test`) were run locally and passed
- [X] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

It will always add `EventEmitter` and `proxyOutputs` imports event if there are no components with event.

Issue URL: Resolves #331

## What is the new behavior?

Include import `EventEmitter` and `proxyOutputs` only if there is at least one component with event.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by liamdebeasi** [comment]
Thanks for the PR! Is there an open issue on GitHub that you can link this to?

**Comment by KariiO** [comment]
> 

Unfortunately there is no issue for that - if its required I'll create one.

**Comment by liamdebeasi** [comment]
Yes, please create one. We require PRs to be linked to triaged issues before they can be merged. Thanks!

**Review by JessicaSachs — APPROVED** [review]
LGTM but would love another set of eyes on it from someone who knows Angular a bit better.

**Comment by liamdebeasi** [comment]
@sean-perkins Mind taking a look when you get a chance?

File: packages/angular-output-target/src/output-angular.ts
Comment (sean-perkins) [inline packages/angular-output-target/src/output-angular.ts:71]: This will need to be updated to filter out internal events. We only require creating the proxy for the event if it is public.

e.g.: https://github.com/ionic-team/stencil-ds-output-targets/blob/main/packages/angular-output-target/src/output-angular.ts#L149

Context from Stencil: https://github.com/ionic-team/stencil/blob/9690a8226424db4bf2da4540bfdf826980c916ac/src/compiler/transformers/component-build-conditionals.ts#L28C1-L30

**Review by sean-perkins — CHANGES_REQUESTED** [review]

File: packages/angular-output-target/src/output-angular.ts
Comment (KariiO) [reply packages/angular-output-target/src/output-angular.ts]: @sean-perkins thanks for pointing this out! 
I applied changes - please kindly check in a spare moment.

**Review by sean-perkins — APPROVED** [review]
This looks good to me, thanks for this contribution and great work 🎉

**Pull request merged**
